### DISCUSSION TITLE
encrypt with private, decrypt with public and vice versa

### DISCUSSION THREAD

**Discussion by aviNrich** [openpgpjs/openpgpjs#1248] [Q&A] [answered]
Hi,
I have 2 parties speaking with each other. I want one party to have the private key only and the other to have the public key only.
`server-a` sends messages to `server-b` it needs to encrypt with private. `server-b` will decrypt the message using the public.
`server-b` sends messages to `server-a` it needs to encrypt with public . `server-a` will decrypt the message using the private.

I know it is doable with RSA, but RSA has performance issues and pgp seems to handle this much faster. 
Is my challenge doable with openpgpjs? maybe something similar? I don't want both parties to have the same string (password, public or private), each party should have only one.

Thanks!

**Comment by twiss** [comment, marked as answer]
Hey :wave:
Asymmetric encryption is always done using the public key, and decryption is always done using the private key.
However, you can generate two key pairs, one for A, and one for B, and give A.private to A, B.private to B, A.public to B, and B.public to A.
Then, A can encrypt with B.public, and B can decrypt with B.private.
And, B can encrypt with A.public, and A can decrypt with A.private.

**Reply by aviNrich** [reply]
Great idea. Work for me. thanks
